packages/esm-appointments-app/src/form/appointments-form.component.tsx

```diff
@@ -195,6 +195,20 @@ const AppointmentsForm: React.FC<AppointmentsFormProps> = ({
 
   useEffect(() => setValue('formIsRecurringAppointment', isRecurringAppointment), [isRecurringAppointment]);
 
+  const {
```

Comment:
would be good to add a brielf comment here as to why we are doing all this

Reply:
@mogoodrich if you mean the useEffect that calls setValue() then that is from a previous PR not this one.

Reply:
I just meant the stuff you added as mentioned in the PR.  Everything looks good to me, I would just include a brief summary here so that someone in the future doesn't refactor this without understanding the reasoning behind it.

Reply:
@mogoodrich We are basically trying to manually call the ref callback from react-hook-form with the DOM node that we want to be focused. Rather than passing it as a ref prop which seems not be working

Reply:
Oh, sorry for not being clear @mccarthyaaron .  When I asked to include a brief summary "here", I meant to add a brief comment to the code itself, so a year or two down the line someone doesn't try to "simplify" the code and remove your changes without understanding why you added them.  Otherwise, this looks good to go!

Reply:
@mogoodrich let me work on it

Reply:
@mogoodrich I was also hoping to get some feedback from @ibacher and @denniskigen regarding this issue

Reply:
This is, unfortunately, basically the right approach. I think it's slightly better to use `getElementById('startDatePickerInput')` vs `querySelector()`, but they're essentially equivalent. Unfortunately, the `ref`s passed directly into the Carbon components point to the div rather than the `input`s (using `ref` pointing to the `DatePicker` div and then calling `ref.querySelector('input:nth-of-type(1)')` and `ref.querySelector('input:nth-of-type(2)`' might be marginally better since it doesn't restrict us from supporting two date editing components on the same page, but I'm not sure we need that flexibility.